## After noticing in the EDA that some of the images contain more than one type of clothing, I decided to analyze the product title with respect to the class label.

In [264]:
import pandas as pd
import numpy as np
import re
import seaborn as sns
import matplotlib.pyplot as plt

In [265]:
df = pd.read_json('/Users/butler/Documents/Image-Classification-Indian-Clothing/data/raw/test_data.json', lines=True)

In [266]:
df.head()

image_url          image_path  \
0  https://m.media-amazon.com/images/I/71NZQ-aaAS...  images/test/0.jpeg   
1  https://m.media-amazon.com/images/I/81aQmbWsG7...  images/test/1.jpeg   
2  https://m.media-amazon.com/images/I/71YjGvzs9y...  images/test/2.jpeg   
3  https://m.media-amazon.com/images/I/81JYhYV5Xr...  images/test/3.jpeg   
4  https://m.media-amazon.com/images/I/91fXc5i5Dy...  images/test/4.jpeg   

                     brand                                      product_title  \
0  TANTUJA BENGAL HANDLOOM  (A Govt. of West Bengal Enterprises) Handloom ...   
1               Trilok Fab               Women's Silk Saree With Blouse Piece   
2              Nine Sister  Women's Georgette Net Embroidered Saree with B...   
3   Gauri Laxmi Enterprise  Women's Crepe Saree with Blouse (GLE$CHICKOO C...   
4                   SOURBH  Women's Cotton Blend Digital Butta Printed Sar...   

  class_label color  
0       saree   NaN  
1       saree   NaN  
2       saree   NaN  
3       saree   NaN  
4       saree   NaN

In [267]:
df.tail()

image_url  \
7495  https://rukminim1.flixcart.com/image/309/371/k...   
7496  https://m.media-amazon.com/images/I/71xJFbz+uV...   
7497  https://m.media-amazon.com/images/I/71CawDGew+...   
7498  https://m.media-amazon.com/images/I/61VBgoJQAt...   
7499  https://m.media-amazon.com/images/I/612gggywBm...   

                 image_path             brand  \
7495  images/test/7495.jpeg    aaliya fashion   
7496  images/test/7496.jpeg  AnjuShree Choice   
7497  images/test/7497.jpeg       W for Woman   
7498  images/test/7498.jpeg            Vaamsi   
7499  images/test/7499.jpeg             Vbuyz   

                                       product_title  class_label   color  
7495          Women Embroidered Rayon Anarkali Kurta  women_kurta  Purple  
7496  Women's Rayon & Viscose Regular Stitched Kurta  women_kurta     NaN  
7497                   Women's cotton straight Kurta  women_kurta     NaN  
7498  Crepe Printed Kurti (VPK1805_Purple_Free Size)  women_kurta     NaN  
7499                     Women's Rayon Regular Kurta  women_kurta     NaN

In [268]:
df.shape

(7500, 6)

In [269]:
df.class_label.unique()

array(['saree', 'blouse', 'dhoti_pants', 'dupattas', 'gowns', 'kurta_men',
       'leggings_and_salwars', 'lehenga', 'mojaris_men', 'mojaris_women',
       'nehru_jackets', 'palazzos', 'petticoats', 'sherwanis',
       'women_kurta'], dtype=object)

In [270]:
# Making a copy of the df
df_revised = df.copy()

In [271]:
df_revised.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7500 entries, 0 to 7499
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   image_url      7500 non-null   object
 1   image_path     7500 non-null   object
 2   brand          7500 non-null   object
 3   product_title  7500 non-null   object
 4   class_label    7500 non-null   object
 5   color          905 non-null    object
dtypes: object(6)
memory usage: 351.7+ KB


In [272]:
# Reset index in case the number of the image is needed later
df_revised = df_revised.reset_index(names='number')

In [273]:
df_revised.head()

number                                          image_url  \
0       0  https://m.media-amazon.com/images/I/71NZQ-aaAS...   
1       1  https://m.media-amazon.com/images/I/81aQmbWsG7...   
2       2  https://m.media-amazon.com/images/I/71YjGvzs9y...   
3       3  https://m.media-amazon.com/images/I/81JYhYV5Xr...   
4       4  https://m.media-amazon.com/images/I/91fXc5i5Dy...   

           image_path                    brand  \
0  images/test/0.jpeg  TANTUJA BENGAL HANDLOOM   
1  images/test/1.jpeg               Trilok Fab   
2  images/test/2.jpeg              Nine Sister   
3  images/test/3.jpeg   Gauri Laxmi Enterprise   
4  images/test/4.jpeg                   SOURBH   

                                       product_title class_label color  
0  (A Govt. of West Bengal Enterprises) Handloom ...       saree   NaN  
1               Women's Silk Saree With Blouse Piece       saree   NaN  
2  Women's Georgette Net Embroidered Saree with B...       saree   NaN  
3  Women's Crepe Saree with Blouse (GLE$CHICKOO C...       saree   NaN  
4  Women's Cotton Blend Digital Butta Printed Sar...       saree   NaN

In [274]:
df_revised.tail()

number                                          image_url  \
7495    7495  https://rukminim1.flixcart.com/image/309/371/k...   
7496    7496  https://m.media-amazon.com/images/I/71xJFbz+uV...   
7497    7497  https://m.media-amazon.com/images/I/71CawDGew+...   
7498    7498  https://m.media-amazon.com/images/I/61VBgoJQAt...   
7499    7499  https://m.media-amazon.com/images/I/612gggywBm...   

                 image_path             brand  \
7495  images/test/7495.jpeg    aaliya fashion   
7496  images/test/7496.jpeg  AnjuShree Choice   
7497  images/test/7497.jpeg       W for Woman   
7498  images/test/7498.jpeg            Vaamsi   
7499  images/test/7499.jpeg             Vbuyz   

                                       product_title  class_label   color  
7495          Women Embroidered Rayon Anarkali Kurta  women_kurta  Purple  
7496  Women's Rayon & Viscose Regular Stitched Kurta  women_kurta     NaN  
7497                   Women's cotton straight Kurta  women_kurta     NaN  
7498  Crepe Printed Kurti (VPK1805_Purple_Free Size)  women_kurta     NaN  
7499                     Women's Rayon Regular Kurta  women_kurta     NaN

In [275]:
# Using column names with smallest part of clothing name
columns = ['kurta', 'women', 'men', 'dhoti', 'saree', 'sherwani', 'palazzo', 
           'mojari', 'nehru', 'legging', 'salwar', 'petticoat', 'blouse', 
           'dupatta', 'gown','lehenga']


In [276]:
# Adding these columns to the df with the values being true if the product title contains the column name
for column in columns:
    df_revised[str(column)] = df_revised.product_title.str.contains(str(column), flags=re.IGNORECASE)

In [277]:
# Checking for rows that have girl in the product title
df_revised['girl'] = df_revised.product_title.str.contains('girl', flags=re.IGNORECASE)

In [278]:
# Putting the rows that have girl in the product title into the women category
c1 = (df_revised.women == False)
c2 = (df_revised.girl == True)

df_revised.loc[c2 & c2, 'women'] = True

In [279]:
# Checking for rows that have woman in the product title
df_revised['woman'] = df_revised.product_title.str.contains('woman', flags=re.IGNORECASE)

In [280]:
# Putting the rows that have woman in the product title into the women category
c1 = (df_revised.women == False)
c2 = (df_revised.woman == True)

df_revised.loc[c2 & c2, 'women'] = True

In [281]:
# Checking for rows that have man in the product title
df_revised['man'] = df_revised.product_title.str.contains('man', flags=re.IGNORECASE)

In [282]:
# Checking for rows that have boy in the product title
df_revised['boy'] = df_revised.product_title.str.contains('boy', flags=re.IGNORECASE)

In [283]:
# Putting the rows that have man in the product title into the men category
c1 = (df_revised.men == False)
c2 = (df_revised.man == True)

df_revised.loc[c2 & c2, 'men'] = True

In [284]:
# Putting the rows that have boy in the product title into the men category
c1 = (df_revised.men == False)
c2 = (df_revised.boy == True)

df_revised.loc[c2 & c2, 'men'] = True

In [285]:
# No longer needed
df_revised.drop(['girl', 'woman', 'man', 'boy'], axis=1, inplace=True)
df_revised.head()

number                                          image_url  \
0       0  https://m.media-amazon.com/images/I/71NZQ-aaAS...   
1       1  https://m.media-amazon.com/images/I/81aQmbWsG7...   
2       2  https://m.media-amazon.com/images/I/71YjGvzs9y...   
3       3  https://m.media-amazon.com/images/I/81JYhYV5Xr...   
4       4  https://m.media-amazon.com/images/I/91fXc5i5Dy...   

           image_path                    brand  \
0  images/test/0.jpeg  TANTUJA BENGAL HANDLOOM   
1  images/test/1.jpeg               Trilok Fab   
2  images/test/2.jpeg              Nine Sister   
3  images/test/3.jpeg   Gauri Laxmi Enterprise   
4  images/test/4.jpeg                   SOURBH   

                                       product_title class_label color  kurta  \
0  (A Govt. of West Bengal Enterprises) Handloom ...       saree   NaN  False   
1               Women's Silk Saree With Blouse Piece       saree   NaN  False   
2  Women's Georgette Net Embroidered Saree with B...       saree   NaN  False   
3  Women's Crepe Saree with Blouse (GLE$CHICKOO C...       saree   NaN  False   
4  Women's Cotton Blend Digital Butta Printed Sar...       saree   NaN  False   

   women   men  ...  palazzo  mojari  nehru  legging  salwar  petticoat  \
0   True  True  ...    False   False  False    False   False      False   
1   True  True  ...    False   False  False    False   False      False   
2   True  True  ...    False   False  False    False   False      False   
3   True  True  ...    False   False  False    False   False      False   
4   True  True  ...    False   False  False    False   False      False   

   blouse  dupatta   gown  lehenga  
0   False    False  False    False  
1    True    False  False    False  
2    True    False  False    False  
3    True    False  False    False  
4    True    False  False    False  

[5 rows x 23 columns]

In [286]:
# If the columns entitled 'men' and 'women' are both true, changing the 'men' column to false
c1 = (df_revised['men'] == True) 
c2 = (df_revised['women'] == True)

df_revised.loc[c1 & c2, 'men'] = False

In [287]:
# Getting the data for which the 'men' column is true now
df_revised_men = df_revised[df_revised.men == True]

In [288]:
# Making a column for women's kurtas
df_revised['women_kurta'] = [False]* 7500

In [289]:
# Making a column for men's kurtas
df_revised['kurta_men'] = [False]* 7500

In [290]:
df_revised.head()

number                                          image_url  \
0       0  https://m.media-amazon.com/images/I/71NZQ-aaAS...   
1       1  https://m.media-amazon.com/images/I/81aQmbWsG7...   
2       2  https://m.media-amazon.com/images/I/71YjGvzs9y...   
3       3  https://m.media-amazon.com/images/I/81JYhYV5Xr...   
4       4  https://m.media-amazon.com/images/I/91fXc5i5Dy...   

           image_path                    brand  \
0  images/test/0.jpeg  TANTUJA BENGAL HANDLOOM   
1  images/test/1.jpeg               Trilok Fab   
2  images/test/2.jpeg              Nine Sister   
3  images/test/3.jpeg   Gauri Laxmi Enterprise   
4  images/test/4.jpeg                   SOURBH   

                                       product_title class_label color  kurta  \
0  (A Govt. of West Bengal Enterprises) Handloom ...       saree   NaN  False   
1               Women's Silk Saree With Blouse Piece       saree   NaN  False   
2  Women's Georgette Net Embroidered Saree with B...       saree   NaN  False   
3  Women's Crepe Saree with Blouse (GLE$CHICKOO C...       saree   NaN  False   
4  Women's Cotton Blend Digital Butta Printed Sar...       saree   NaN  False   

   women    men  ...  nehru  legging  salwar  petticoat  blouse  dupatta  \
0   True  False  ...  False    False   False      False   False    False   
1   True  False  ...  False    False   False      False    True    False   
2   True  False  ...  False    False   False      False    True    False   
3   True  False  ...  False    False   False      False    True    False   
4   True  False  ...  False    False   False      False    True    False   

    gown  lehenga  women_kurta  kurta_men  
0  False    False        False      False  
1  False    False        False      False  
2  False    False        False      False  
3  False    False        False      False  
4  False    False        False      False  

[5 rows x 25 columns]

In [291]:
# Making women_kurta true and kurta false if kurta is true and women is true
c1 = (df_revised['kurta'] == True) 
c2 = (df_revised['women'] == True)

df_revised.loc[c1 & c2, 'women_kurta'] = True
df_revised.loc[c1 & c2, 'kurta'] = False

In [292]:
# Making kurta_men true and kurta false if kurta is true and men is true
c1 = (df_revised['kurta'] == True) 
c2 = (df_revised['men'] == True)

df_revised.loc[c1 & c2, 'kurta_men'] = True
df_revised.loc[c1 & c2, 'kurta'] = False

In [293]:
# Since only women wear dupattas, if the class_label is dupatta and kurta is true, is must
# be a women's kurta
c1 = (df_revised.class_label == 'dupattas')
c2 = (df_revised.kurta == True)

df_revised.loc[c1 & c2, 'women_kurta'] = True
df_revised.loc[c1 & c2, 'kurta'] = False

In [294]:
# Making kurta false when if kurta is true and kurta_men is the class label
c1 = (df_revised['kurta'] == True) 
c2 = (df_revised.class_label == 'kurta_men')

df_revised.loc[c1 & c2, 'kurta'] = False

In [295]:
# Making kurta false when if kurta is true and women_kurta is the class label
c1 = (df_revised['kurta'] == True) 
c2 = (df_revised.class_label == 'women_kurta')

df_revised.loc[c1 & c2, 'kurta'] = False

In [296]:
df_revised.kurta.value_counts()

kurta
False    7484
True       16
Name: count, dtype: int64

In [297]:
(df_revised.kurta == 'kurta').value_counts()

kurta
False    7500
Name: count, dtype: int64

## This leaves 16 rows where kurta is true, men and women are false, and the class label does not contain kurta

In [298]:
# Making columns for mojaris_men, mojaris_women, and leggings_and_salwars
df_revised['mojaris_men'] = [False]*7500
df_revised['mojaris_women'] = [False]*7500
df_revised['leggings_and_salwars'] = [False]*7500

In [299]:
# mojaris_women column is true if mojaris and women are true
c1 = (df_revised['mojari'] == True) 
c2 = (df_revised['women'] == True)

df_revised.loc[c1 & c2, 'mojaris_women'] = True

In [300]:
# mojaris_men column is true if mojaris and men are true
c1 = (df_revised['mojari'] == True) 
c2 = (df_revised['men'] == True)

df_revised.loc[c1 & c2, 'mojaris_men'] = True

In [301]:
# Putting the columns legging and salwar together
c1 = (df_revised['legging'] == True) 
c2 = (df_revised['salwar'] == True)

df_revised.loc[c1 | c2, 'leggings_and_salwars'] = True

In [302]:
# Dropping the columns that are no longer needed
df_revised.drop(['men', 'women', 'legging','salwar','mojari'], axis=1, inplace=True)

In [303]:
df_revised.head()

number                                          image_url  \
0       0  https://m.media-amazon.com/images/I/71NZQ-aaAS...   
1       1  https://m.media-amazon.com/images/I/81aQmbWsG7...   
2       2  https://m.media-amazon.com/images/I/71YjGvzs9y...   
3       3  https://m.media-amazon.com/images/I/81JYhYV5Xr...   
4       4  https://m.media-amazon.com/images/I/91fXc5i5Dy...   

           image_path                    brand  \
0  images/test/0.jpeg  TANTUJA BENGAL HANDLOOM   
1  images/test/1.jpeg               Trilok Fab   
2  images/test/2.jpeg              Nine Sister   
3  images/test/3.jpeg   Gauri Laxmi Enterprise   
4  images/test/4.jpeg                   SOURBH   

                                       product_title class_label color  kurta  \
0  (A Govt. of West Bengal Enterprises) Handloom ...       saree   NaN  False   
1               Women's Silk Saree With Blouse Piece       saree   NaN  False   
2  Women's Georgette Net Embroidered Saree with B...       saree   NaN  False   
3  Women's Crepe Saree with Blouse (GLE$CHICKOO C...       saree   NaN  False   
4  Women's Cotton Blend Digital Butta Printed Sar...       saree   NaN  False   

   dhoti  saree  ...  petticoat  blouse  dupatta   gown  lehenga  women_kurta  \
0  False  False  ...      False   False    False  False    False        False   
1  False   True  ...      False    True    False  False    False        False   
2  False   True  ...      False    True    False  False    False        False   
3  False   True  ...      False    True    False  False    False        False   
4  False   True  ...      False    True    False  False    False        False   

   kurta_men  mojaris_men  mojaris_women  leggings_and_salwars  
0      False        False          False                 False  
1      False        False          False                 False  
2      False        False          False                 False  
3      False        False          False                 False  
4      False        False          False                 False  

[5 rows x 23 columns]

In [304]:
# Removing the blouse label from all sarees in class_label because 
# models wearing sarees cannot be shown without a saree blouse

c1 = (df_revised['class_label'] == 'saree') 
c2 = (df_revised['blouse'] == True)

df_revised.loc[c1 & c2, 'blouse'] = False

In [305]:
# Removing the saree label from all blouses in class_label because 
# blouses are blouses for sarees or lehengas. Therefore, it is likely that saree 
# will be in the product description

c1 = (df_revised['class_label'] == 'blouse') 
c2 = (df_revised['saree'] == True)

df_revised.loc[c1 & c2, 'saree'] = False

In [306]:
# Removing the saree label from all blouses in class_label because 
# blouses are blouses for sarees or lehengas. Therefore, it is likely that lehenga 
# will be in the product description

c1 = (df_revised['class_label'] == 'blouse') 
c2 = (df_revised['lehenga'] == True)

df_revised.loc[c1 & c2, 'lehenga'] = False

In [307]:
# Gowns are sometimes called dresses
df_revised['dress'] = df_revised.product_title.str.contains('dress', flags=re.IGNORECASE)

In [308]:
# If dress is true and gowns is not, put the value for gowns as True in gown column
c1 = (df_revised.gown == False)
c2 = (df_revised.dress == True)

df_revised.loc[c1 & c2, 'gown'] = True

In [309]:
df_revised.head()

number                                          image_url  \
0       0  https://m.media-amazon.com/images/I/71NZQ-aaAS...   
1       1  https://m.media-amazon.com/images/I/81aQmbWsG7...   
2       2  https://m.media-amazon.com/images/I/71YjGvzs9y...   
3       3  https://m.media-amazon.com/images/I/81JYhYV5Xr...   
4       4  https://m.media-amazon.com/images/I/91fXc5i5Dy...   

           image_path                    brand  \
0  images/test/0.jpeg  TANTUJA BENGAL HANDLOOM   
1  images/test/1.jpeg               Trilok Fab   
2  images/test/2.jpeg              Nine Sister   
3  images/test/3.jpeg   Gauri Laxmi Enterprise   
4  images/test/4.jpeg                   SOURBH   

                                       product_title class_label color  kurta  \
0  (A Govt. of West Bengal Enterprises) Handloom ...       saree   NaN  False   
1               Women's Silk Saree With Blouse Piece       saree   NaN  False   
2  Women's Georgette Net Embroidered Saree with B...       saree   NaN  False   
3  Women's Crepe Saree with Blouse (GLE$CHICKOO C...       saree   NaN  False   
4  Women's Cotton Blend Digital Butta Printed Sar...       saree   NaN  False   

   dhoti  saree  ...  blouse  dupatta   gown  lehenga  women_kurta  kurta_men  \
0  False  False  ...   False    False  False    False        False      False   
1  False   True  ...   False    False  False    False        False      False   
2  False   True  ...   False    False  False    False        False      False   
3  False   True  ...   False    False  False    False        False      False   
4  False   True  ...   False    False  False    False        False      False   

   mojaris_men  mojaris_women  leggings_and_salwars  dress  
0        False          False                 False  False  
1        False          False                 False  False  
2        False          False                 False  False  
3        False          False                 False  False  
4        False          False                 False  False  

[5 rows x 24 columns]

In [310]:
df_revised.loc[1]

number                                                                  1
image_url               https://m.media-amazon.com/images/I/81aQmbWsG7...
image_path                                             images/test/1.jpeg
brand                                                          Trilok Fab
product_title                        Women's Silk Saree With Blouse Piece
class_label                                                         saree
color                                                                 NaN
kurta                                                               False
dhoti                                                               False
saree                                                                True
sherwani                                                            False
palazzo                                                             False
nehru                                                               False
petticoat                             

In [311]:
# Renaming columns to conform to class labels
df_revised = df_revised.rename({'dhoti':'dhoti_pants', 'sherwani':'sherwanis', 'palazzo':'palazzos', 
             'nehru':'nehru_jackets', 'petticoat':'petticoats', 'dupatta':'dupattas', 
             'gown':'gowns'},axis=1)
df_revised.head()

number                                          image_url  \
0       0  https://m.media-amazon.com/images/I/71NZQ-aaAS...   
1       1  https://m.media-amazon.com/images/I/81aQmbWsG7...   
2       2  https://m.media-amazon.com/images/I/71YjGvzs9y...   
3       3  https://m.media-amazon.com/images/I/81JYhYV5Xr...   
4       4  https://m.media-amazon.com/images/I/91fXc5i5Dy...   

           image_path                    brand  \
0  images/test/0.jpeg  TANTUJA BENGAL HANDLOOM   
1  images/test/1.jpeg               Trilok Fab   
2  images/test/2.jpeg              Nine Sister   
3  images/test/3.jpeg   Gauri Laxmi Enterprise   
4  images/test/4.jpeg                   SOURBH   

                                       product_title class_label color  kurta  \
0  (A Govt. of West Bengal Enterprises) Handloom ...       saree   NaN  False   
1               Women's Silk Saree With Blouse Piece       saree   NaN  False   
2  Women's Georgette Net Embroidered Saree with B...       saree   NaN  False   
3  Women's Crepe Saree with Blouse (GLE$CHICKOO C...       saree   NaN  False   
4  Women's Cotton Blend Digital Butta Printed Sar...       saree   NaN  False   

   dhoti_pants  saree  ...  blouse  dupattas  gowns  lehenga  women_kurta  \
0        False  False  ...   False     False  False    False        False   
1        False   True  ...   False     False  False    False        False   
2        False   True  ...   False     False  False    False        False   
3        False   True  ...   False     False  False    False        False   
4        False   True  ...   False     False  False    False        False   

   kurta_men  mojaris_men  mojaris_women  leggings_and_salwars  dress  
0      False        False          False                 False  False  
1      False        False          False                 False  False  
2      False        False          False                 False  False  
3      False        False          False                 False  False  
4      False        False          False                 False  False  

[5 rows x 24 columns]

In [312]:
# Changing each new column to false if the class_label is the same as the column name
columns = ['dhoti_pants', 'saree', 'sherwanis', 'palazzos', 'nehru_jackets', 'petticoats',
           'blouse','dupattas', 'gowns', 'lehenga', 'women_kurta', 'kurta_men', 
           'mojaris_men', 'mojaris_women', 'leggings_and_salwars']

for column in columns:
    c1 = (df_revised[str(column)] == True)
    c2 = (df_revised['class_label'] == str(column))
    df_revised.loc[c1 & c2, str(column)] = False

df_revised.head()

number                                          image_url  \
0       0  https://m.media-amazon.com/images/I/71NZQ-aaAS...   
1       1  https://m.media-amazon.com/images/I/81aQmbWsG7...   
2       2  https://m.media-amazon.com/images/I/71YjGvzs9y...   
3       3  https://m.media-amazon.com/images/I/81JYhYV5Xr...   
4       4  https://m.media-amazon.com/images/I/91fXc5i5Dy...   

           image_path                    brand  \
0  images/test/0.jpeg  TANTUJA BENGAL HANDLOOM   
1  images/test/1.jpeg               Trilok Fab   
2  images/test/2.jpeg              Nine Sister   
3  images/test/3.jpeg   Gauri Laxmi Enterprise   
4  images/test/4.jpeg                   SOURBH   

                                       product_title class_label color  kurta  \
0  (A Govt. of West Bengal Enterprises) Handloom ...       saree   NaN  False   
1               Women's Silk Saree With Blouse Piece       saree   NaN  False   
2  Women's Georgette Net Embroidered Saree with B...       saree   NaN  False   
3  Women's Crepe Saree with Blouse (GLE$CHICKOO C...       saree   NaN  False   
4  Women's Cotton Blend Digital Butta Printed Sar...       saree   NaN  False   

   dhoti_pants  saree  ...  blouse  dupattas  gowns  lehenga  women_kurta  \
0        False  False  ...   False     False  False    False        False   
1        False  False  ...   False     False  False    False        False   
2        False  False  ...   False     False  False    False        False   
3        False  False  ...   False     False  False    False        False   
4        False  False  ...   False     False  False    False        False   

   kurta_men  mojaris_men  mojaris_women  leggings_and_salwars  dress  
0      False        False          False                 False  False  
1      False        False          False                 False  False  
2      False        False          False                 False  False  
3      False        False          False                 False  False  
4      False        False          False                 False  False  

[5 rows x 24 columns]

In [313]:
columns = ['dhoti_pants', 'saree', 'kurta', 'sherwanis', 'palazzos', 'nehru_jackets', 'petticoats',
           'blouse','dupattas', 'gowns', 'lehenga', 'women_kurta', 'kurta_men', 
           'mojaris_men', 'mojaris_women', 'leggings_and_salwars']
# If a column is true, putting the column name as the value; otherwise, 0
def func(x):
    if x == True:
        return str(column)
    else:
        return 0

for column in columns:
    df_revised[str(column)] = df_revised[str(column)].apply(lambda x: func(x))

In [314]:
df_revised.head()

number                                          image_url  \
0       0  https://m.media-amazon.com/images/I/71NZQ-aaAS...   
1       1  https://m.media-amazon.com/images/I/81aQmbWsG7...   
2       2  https://m.media-amazon.com/images/I/71YjGvzs9y...   
3       3  https://m.media-amazon.com/images/I/81JYhYV5Xr...   
4       4  https://m.media-amazon.com/images/I/91fXc5i5Dy...   

           image_path                    brand  \
0  images/test/0.jpeg  TANTUJA BENGAL HANDLOOM   
1  images/test/1.jpeg               Trilok Fab   
2  images/test/2.jpeg              Nine Sister   
3  images/test/3.jpeg   Gauri Laxmi Enterprise   
4  images/test/4.jpeg                   SOURBH   

                                       product_title class_label color kurta  \
0  (A Govt. of West Bengal Enterprises) Handloom ...       saree   NaN     0   
1               Women's Silk Saree With Blouse Piece       saree   NaN     0   
2  Women's Georgette Net Embroidered Saree with B...       saree   NaN     0   
3  Women's Crepe Saree with Blouse (GLE$CHICKOO C...       saree   NaN     0   
4  Women's Cotton Blend Digital Butta Printed Sar...       saree   NaN     0   

  dhoti_pants saree  ... blouse dupattas gowns lehenga women_kurta kurta_men  \
0           0     0  ...      0        0     0       0           0         0   
1           0     0  ...      0        0     0       0           0         0   
2           0     0  ...      0        0     0       0           0         0   
3           0     0  ...      0        0     0       0           0         0   
4           0     0  ...      0        0     0       0           0         0   

  mojaris_men mojaris_women leggings_and_salwars  dress  
0           0             0                    0  False  
1           0             0                    0  False  
2           0             0                    0  False  
3           0             0                    0  False  
4           0             0                    0  False  

[5 rows x 24 columns]

In [315]:
df_revised.tail()

number                                          image_url  \
7495    7495  https://rukminim1.flixcart.com/image/309/371/k...   
7496    7496  https://m.media-amazon.com/images/I/71xJFbz+uV...   
7497    7497  https://m.media-amazon.com/images/I/71CawDGew+...   
7498    7498  https://m.media-amazon.com/images/I/61VBgoJQAt...   
7499    7499  https://m.media-amazon.com/images/I/612gggywBm...   

                 image_path             brand  \
7495  images/test/7495.jpeg    aaliya fashion   
7496  images/test/7496.jpeg  AnjuShree Choice   
7497  images/test/7497.jpeg       W for Woman   
7498  images/test/7498.jpeg            Vaamsi   
7499  images/test/7499.jpeg             Vbuyz   

                                       product_title  class_label   color  \
7495          Women Embroidered Rayon Anarkali Kurta  women_kurta  Purple   
7496  Women's Rayon & Viscose Regular Stitched Kurta  women_kurta     NaN   
7497                   Women's cotton straight Kurta  women_kurta     NaN   
7498  Crepe Printed Kurti (VPK1805_Purple_Free Size)  women_kurta     NaN   
7499                     Women's Rayon Regular Kurta  women_kurta     NaN   

     kurta dhoti_pants saree  ... blouse dupattas gowns lehenga women_kurta  \
7495     0           0     0  ...      0        0     0       0           0   
7496     0           0     0  ...      0        0     0       0           0   
7497     0           0     0  ...      0        0     0       0           0   
7498     0           0     0  ...      0        0     0       0           0   
7499     0           0     0  ...      0        0     0       0           0   

     kurta_men mojaris_men mojaris_women leggings_and_salwars  dress  
7495         0           0             0                    0  False  
7496         0           0             0                    0  False  
7497         0           0             0                    0  False  
7498         0           0             0                    0  False  
7499         0           0             0                    0  False  

[5 rows x 24 columns]

In [316]:
# Removing columns not needed
df_revised = df_revised.drop(['number', 'image_url', 'brand', 'color', 'dress'], axis=1)

In [317]:
# Saving revised dataframe
df_revised.to_csv('/Users/butler/Documents/Image-Classification-Indian-Clothing/data/interim/test_other_categories.csv')

### Making a dataframe for each label that is different than the class label and exploring the product titles and class labels of some of the rows

### Sarees

In [318]:
df_saree = df_revised[df_revised.saree == 'saree']
df_saree.shape

(191, 19)

In [319]:
df_saree.head()

image_path  \
2056  images/test/2056.jpeg   
2090  images/test/2090.jpeg   
2265  images/test/2265.jpeg   
2397  images/test/2397.jpeg   
2412  images/test/2412.jpeg   

                                          product_title class_label kurta  \
2056  Saree for women | Women's Fancy Saree | Silk S...       gowns     0   
2090  Chiffon with Blouse Piece Saree (Riva516_Yello...       gowns     0   
2265  DaSync Silver Stone Hanging latkan Laces Borde...       gowns     0   
2397                    Women's Saree with blouse piece       gowns     0   
2412              Women's Satin Saree With Blouse Piece       gowns     0   

     dhoti_pants  saree sherwanis palazzos nehru_jackets petticoats  blouse  \
2056           0  saree         0        0             0          0  blouse   
2090           0  saree         0        0             0          0  blouse   
2265           0  saree         0        0             0          0  blouse   
2397           0  saree         0        0             0          0  blouse   
2412           0  saree         0        0             0          0  blouse   

     dupattas gowns  lehenga women_kurta kurta_men mojaris_men mojaris_women  \
2056        0     0        0           0         0           0             0   
2090        0     0        0           0         0           0             0   
2265        0     0  lehenga           0         0           0             0   
2397        0     0        0           0         0           0             0   
2412        0     0        0           0         0           0             0   

     leggings_and_salwars  
2056                    0  
2090                    0  
2265                    0  
2397                    0  
2412                    0

### Mislabeled:

In [320]:
print(f'Product title: {df_saree.iloc[0,1]}')
print(f'Class label: {df_saree.iloc[0,2]}')

Product title: Saree for women | Women's Fancy Saree | Silk Saree With Silk Blouse Women's Saree (PARENT-ZQ-152)
Class label: gowns


In [321]:
print(f'Product title: {df_saree.iloc[3,1]}')
print(f'Class label: {df_saree.iloc[3,2]}')

Product title: Women's Saree with blouse piece
Class label: gowns


In [322]:
print(f'Product title: {df_saree.iloc[4,1]}')
print(f'Class label: {df_saree.iloc[4,2]}')

Product title: Women's Satin Saree With Blouse Piece
Class label: gowns


### Blouses

In [323]:
df_blouse = df_revised[df_revised.blouse == 'blouse']
df_blouse.shape

(201, 19)

In [324]:
df_blouse.head()

image_path  \
1769  images/test/1769.jpeg   
2056  images/test/2056.jpeg   
2069  images/test/2069.jpeg   
2090  images/test/2090.jpeg   
2115  images/test/2115.jpeg   

                                          product_title class_label kurta  \
1769  Women'S Kashmiri Art Printed Shawl Dupatta Wit...    dupattas     0   
2056  Saree for women | Women's Fancy Saree | Silk S...       gowns     0   
2069  Chlamys Women Floral Printed Crape Unstiched F...       gowns     0   
2090  Chiffon with Blouse Piece Saree (Riva516_Yello...       gowns     0   
2115  Woman's/Girl's Ethnic Unstitched Art Crepe Pri...       gowns     0   

     dhoti_pants  saree sherwanis  palazzos nehru_jackets petticoats  blouse  \
1769           0      0         0         0             0          0  blouse   
2056           0  saree         0         0             0          0  blouse   
2069           0      0         0         0             0          0  blouse   
2090           0  saree         0         0             0          0  blouse   
2115           0      0         0  palazzos             0          0  blouse   

     dupattas gowns lehenga women_kurta kurta_men mojaris_men mojaris_women  \
1769        0     0       0           0         0           0             0   
2056        0     0       0           0         0           0             0   
2069        0     0       0           0         0           0             0   
2090        0     0       0           0         0           0             0   
2115        0     0       0           0         0           0             0   

     leggings_and_salwars  
1769                    0  
2056                    0  
2069                    0  
2090                    0  
2115                    0

### Mislabeled:

In [325]:
print(f'Product title: {df_blouse.iloc[3,1]}')
print(f'Class label: {df_blouse.iloc[3,2]}')

Product title: Chiffon with Blouse Piece Saree (Riva516_Yellow_Free Size)
Class label: gowns


### Dhoti Pants

In [326]:
df_dhoti_pants = df_revised[df_revised.dhoti_pants == 'dhoti_pants']
df_dhoti_pants.shape

(45, 19)

In [327]:
df_dhoti_pants.head()

image_path  \
117    images/test/117.jpeg   
2328  images/test/2328.jpeg   
2541  images/test/2541.jpeg   
2735  images/test/2735.jpeg   
2782  images/test/2782.jpeg   

                                          product_title class_label kurta  \
117   Women's Olive Green Poly Crepe Crop Top And Dh...       saree     0   
2328                    Women's Regular Fit Dhoti Pants       gowns     0   
2541                              Men Beige Dhoti Pants   kurta_men     0   
2735         Men Beige Dupion Silk Dhoti & Maroon Kurta   kurta_men     0   
2782             Men Green Solid Kurta with Dhoti Pants   kurta_men     0   

      dhoti_pants saree sherwanis palazzos nehru_jackets petticoats blouse  \
117   dhoti_pants     0         0        0             0          0      0   
2328  dhoti_pants     0         0        0             0          0      0   
2541  dhoti_pants     0         0        0             0          0      0   
2735  dhoti_pants     0         0        0             0          0      0   
2782  dhoti_pants     0         0        0             0          0      0   

      dupattas gowns lehenga women_kurta kurta_men mojaris_men mojaris_women  \
117   dupattas     0       0           0         0           0             0   
2328         0     0       0           0         0           0             0   
2541         0     0       0           0         0           0             0   
2735         0     0       0           0         0           0             0   
2782         0     0       0           0         0           0             0   

     leggings_and_salwars  
117                     0  
2328                    0  
2541                    0  
2735                    0  
2782                    0

### Mislabeled:

In [328]:
print(f'Product title: {df_dhoti_pants.iloc[1,1]}')
print(f'Class label: {df_dhoti_pants.iloc[1,2]}')

Product title: Women's Regular Fit Dhoti Pants
Class label: gowns


In [329]:
print(f'Product title: {df_dhoti_pants.iloc[2,1]}')
print(f'Class label: {df_dhoti_pants.iloc[2,2]}')

Product title: Men Beige Dhoti Pants
Class label: kurta_men


### Dupattas

In [330]:
df_dupattas = df_revised[df_revised.dupattas == 'dupattas']
df_dupattas.shape

(278, 19)

In [331]:
df_dupattas.head()

image_path  \
117    images/test/117.jpeg   
262    images/test/262.jpeg   
1002  images/test/1002.jpeg   
1017  images/test/1017.jpeg   
1032  images/test/1032.jpeg   

                                          product_title  class_label kurta  \
117   Women's Olive Green Poly Crepe Crop Top And Dh...        saree     0   
262              Women's Jacquard Banarasi Silk Dupatta        saree     0   
1002  Women Cotton Floral and Embroidered Pattern St...  dhoti_pants     0   
1017  Women’s Pure Cotton Plain Semi Patiala Dhoti S...  dhoti_pants     0   
1032  Women’s Pure Cotton Plain Semi Patiala Dhoti S...  dhoti_pants     0   

      dhoti_pants saree sherwanis palazzos nehru_jackets petticoats blouse  \
117   dhoti_pants     0         0        0             0          0      0   
262             0     0         0        0             0          0      0   
1002            0     0         0        0             0          0      0   
1017            0     0         0        0             0          0      0   
1032            0     0         0        0             0          0      0   

      dupattas gowns lehenga  women_kurta kurta_men mojaris_men mojaris_women  \
117   dupattas     0       0            0         0           0             0   
262   dupattas     0       0            0         0           0             0   
1002  dupattas     0       0  women_kurta         0           0             0   
1017  dupattas     0       0            0         0           0             0   
1032  dupattas     0       0            0         0           0             0   

      leggings_and_salwars  
117                      0  
262                      0  
1002                     0  
1017  leggings_and_salwars  
1032  leggings_and_salwars

### Mislabeled:

In [332]:
print(f'Product title: {df_dupattas.iloc[0,1]}')
print(f'Class label: {df_dupattas.iloc[0,2]}')

Product title: Women's Olive Green Poly Crepe Crop Top And Dhoti With Attached Dupatta
Class label: saree


In [333]:
print(f'Product title: {df_dupattas.iloc[1,1]}')
print(f'Class label: {df_dupattas.iloc[1,2]}')

Product title: Women's Jacquard Banarasi Silk Dupatta
Class label: saree


### Gowns

In [334]:
df_gowns = df_revised[df_revised.gowns == 'gowns']
df_gowns.shape

(63, 19)

In [335]:
df_gowns.head()

image_path  \
568    images/test/568.jpeg   
777    images/test/777.jpeg   
1211  images/test/1211.jpeg   
1341  images/test/1341.jpeg   
1533  images/test/1533.jpeg   

                                          product_title  class_label kurta  \
568   Women/Girls Ethnic Unstitched Chanderi Cotton ...       blouse     0   
777   Women/Girls Ethnic Unstitched Chanderi Cotton ...       blouse     0   
1211  Women's Satin 2 Piece Night Suit Dress (Black,...  dhoti_pants     0   
1341  Girls Western Dresses Cotton Blend Top and Den...  dhoti_pants     0   
1533  Woman's Jacquard Lining Designed Net Dupatta w...     dupattas     0   

     dhoti_pants saree sherwanis palazzos nehru_jackets petticoats blouse  \
568            0     0         0        0             0          0      0   
777            0     0         0        0             0          0      0   
1211           0     0         0        0             0          0      0   
1341           0     0         0        0             0          0      0   
1533           0     0         0        0             0          0      0   

     dupattas  gowns lehenga women_kurta kurta_men mojaris_men mojaris_women  \
568         0  gowns       0           0         0           0             0   
777         0  gowns       0           0         0           0             0   
1211        0  gowns       0           0         0           0             0   
1341        0  gowns       0           0         0           0             0   
1533        0  gowns       0           0         0           0             0   

     leggings_and_salwars  
568                     0  
777                     0  
1211                    0  
1341                    0  
1533                    0

In [336]:
print(f'Product title: {df_gowns.iloc[2,1]}')
print(f'Class label: {df_gowns.iloc[2,2]}')

Product title: Women's Satin 2 Piece Night Suit Dress (Black, Free Size)
Class label: dhoti_pants


### Kurtas

In [337]:
df_kurta = df_revised[df_revised.kurta == 'kurta']
df_kurta.shape

(16, 19)

In [338]:
df_kurta.head()

image_path  \
1181  images/test/1181.jpeg   
1335  images/test/1335.jpeg   
1379  images/test/1379.jpeg   
2107  images/test/2107.jpeg   
2239  images/test/2239.jpeg   

                                          product_title  class_label  kurta  \
1181  Dupion Silk Full Sleeve Long Kurta Blended Dho...  dhoti_pants  kurta   
1335  Dupion Silk Full Sleeve Long Kurta Blended Dho...  dhoti_pants  kurta   
1379  Black Chanderi Silk Kurta with organza hand wo...  dhoti_pants  kurta   
2107                           Impex Ancle Length Kurta        gowns  kurta   
2239  Faux Georgette Embroidered Kurta & Palazzo Mat...        gowns  kurta   

     dhoti_pants saree sherwanis  palazzos nehru_jackets petticoats blouse  \
1181           0     0         0         0             0          0      0   
1335           0     0         0         0             0          0      0   
1379           0     0         0         0             0          0      0   
2107           0     0         0         0             0          0      0   
2239           0     0         0  palazzos             0          0      0   

      dupattas gowns lehenga women_kurta kurta_men mojaris_men mojaris_women  \
1181         0     0       0           0         0           0             0   
1335         0     0       0           0         0           0             0   
1379  dupattas     0       0           0         0           0             0   
2107         0     0       0           0         0           0             0   
2239         0     0       0           0         0           0             0   

     leggings_and_salwars  
1181                    0  
1335                    0  
1379                    0  
2107                    0  
2239                    0

### Mislabeled:

In [339]:
print(f'Product title: {df_kurta.iloc[3,1]}')
print(f'Class label: {df_kurta.iloc[3,2]}')

Product title: Impex Ancle Length Kurta
Class label: gowns


### Men's Kurtas

In [340]:
df_kurta_men = df_revised[df_revised.kurta_men == 'kurta_men']
df_kurta_men.shape

(61, 19)

In [341]:
df_kurta_men.head()

image_path  \
1120  images/test/1120.jpeg   
1222  images/test/1222.jpeg   
1235  images/test/1235.jpeg   
1252  images/test/1252.jpeg   
1269  images/test/1269.jpeg   

                                          product_title  class_label kurta  \
1120  Men's Ethnic Rayon Side Button Long Straight K...  dhoti_pants     0   
1222  Baap Beta Maroon Kurta and Dhoti Pant Set (Man...  dhoti_pants     0   
1235  Mens Silk Self Design Kurta And Dhoti Pant (Blue)  dhoti_pants     0   
1252                                Men's Regular Kurta  dhoti_pants     0   
1269          Mens Self Design Kurta and Dhoti Pant Set  dhoti_pants     0   

     dhoti_pants saree sherwanis palazzos nehru_jackets petticoats blouse  \
1120           0     0         0        0             0          0      0   
1222           0     0         0        0             0          0      0   
1235           0     0         0        0             0          0      0   
1252           0     0         0        0             0          0      0   
1269           0     0         0        0             0          0      0   

     dupattas gowns lehenga women_kurta  kurta_men mojaris_men mojaris_women  \
1120        0     0       0           0  kurta_men           0             0   
1222        0     0       0           0  kurta_men           0             0   
1235        0     0       0           0  kurta_men           0             0   
1252        0     0       0           0  kurta_men           0             0   
1269        0     0       0           0  kurta_men           0             0   

     leggings_and_salwars  
1120                    0  
1222                    0  
1235                    0  
1252                    0  
1269                    0

### Mislabeled:

In [342]:
print(f'Product title: {df_kurta_men.iloc[0,1]}')
print(f'Class label: {df_kurta_men.iloc[0,2]}')

Product title: Men's Ethnic Rayon Side Button Long Straight Kurta
Class label: dhoti_pants


In [343]:
print(f'Product title: {df_kurta_men.iloc[3,1]}')
print(f'Class label: {df_kurta_men.iloc[3,2]}')

Product title: Men's Regular Kurta
Class label: dhoti_pants


### Leggings and Salwars

In [344]:
df_leggings_and_salwars = df_revised[df_revised.leggings_and_salwars == 'leggings_and_salwars']
df_leggings_and_salwars.shape

(214, 19)

In [345]:
df_leggings_and_salwars.head()

image_path  \
234    images/test/234.jpeg   
1003  images/test/1003.jpeg   
1004  images/test/1004.jpeg   
1007  images/test/1007.jpeg   
1017  images/test/1017.jpeg   

                                          product_title  class_label kurta  \
234   The Boutique House - Soft malmal cotton sarees...        saree     0   
1003  Women's Lycra Side Plated Dhoti Patiala Salwar...  dhoti_pants     0   
1004  Women's Cotton Viscose Lycra Dhoti Patiyala Sa...  dhoti_pants     0   
1007  Women's Viscose Lycra Cotton Dhoti Patiyala Sa...  dhoti_pants     0   
1017  Women’s Pure Cotton Plain Semi Patiala Dhoti S...  dhoti_pants     0   

     dhoti_pants saree sherwanis palazzos nehru_jackets petticoats blouse  \
234            0     0         0        0             0          0      0   
1003           0     0         0        0             0          0      0   
1004           0     0         0        0             0          0      0   
1007           0     0         0        0             0          0      0   
1017           0     0         0        0             0          0      0   

      dupattas gowns lehenga women_kurta kurta_men mojaris_men mojaris_women  \
234          0     0       0           0         0           0             0   
1003         0     0       0           0         0           0             0   
1004         0     0       0           0         0           0             0   
1007         0     0       0           0         0           0             0   
1017  dupattas     0       0           0         0           0             0   

      leggings_and_salwars  
234   leggings_and_salwars  
1003  leggings_and_salwars  
1004  leggings_and_salwars  
1007  leggings_and_salwars  
1017  leggings_and_salwars

In [346]:
print(f'Product title: {df_leggings_and_salwars.iloc[0,1]}')
print(f'Class label: {df_leggings_and_salwars.iloc[0,2]}')

Product title: The Boutique House - Soft malmal cotton sarees with zari checks with embroidery Work - Kota Doria cotton saree - Kota Doria Lehanga - Salwar suits - Ethnic Wear - Pattern 3
Class label: saree


### Lehenga

In [347]:
df_lehenga = df_revised[df_revised.lehenga == 'lehenga']
df_lehenga.shape

(29, 19)

In [348]:
df_lehenga.head()

image_path  \
1522  images/test/1522.jpeg   
1527  images/test/1527.jpeg   
1535  images/test/1535.jpeg   
1646  images/test/1646.jpeg   
1665  images/test/1665.jpeg   

                                          product_title class_label kurta  \
1522  Women and Girls Latest Design Chanderi Dupatta...    dupattas     0   
1527  Girl's Embroidered Long Jacket, Lehenga and Du...    dupattas     0   
1535  Women's Silk semi-stitched Lehenga Choli with ...    dupattas     0   
1646  Celebrity Women's Net Embroidered Lehenga Chol...    dupattas     0   
1665  Digital Printed Chandei Latest Floral Design D...    dupattas     0   

     dhoti_pants saree sherwanis palazzos nehru_jackets petticoats blouse  \
1522           0     0         0        0             0          0      0   
1527           0     0         0        0             0          0      0   
1535           0     0         0        0             0          0      0   
1646           0     0         0        0             0          0      0   
1665           0     0         0        0             0          0      0   

     dupattas gowns  lehenga  women_kurta kurta_men mojaris_men mojaris_women  \
1522        0     0  lehenga  women_kurta         0           0             0   
1527        0     0  lehenga            0         0           0             0   
1535        0     0  lehenga            0         0           0             0   
1646        0     0  lehenga            0         0           0             0   
1665        0     0  lehenga  women_kurta         0           0             0   

     leggings_and_salwars  
1522                    0  
1527                    0  
1535                    0  
1646                    0  
1665                    0

In [349]:
print(f'Product title: {df_lehenga.iloc[2,1]}')
print(f'Class label: {df_lehenga.iloc[2,2]}')

Product title: Women's Silk semi-stitched Lehenga Choli with Dupatta (Guldasta1K, Pink, Free Size)
Class label: dupattas


### Men's Mojaris

In [350]:
df_mojaris_men = df_revised[df_revised.mojaris_men == 'mojaris_men']
df_mojaris_men.shape

(16, 19)

In [351]:
df_mojaris_men.head()

image_path  \
4559  images/test/4559.jpeg   
4564  images/test/4564.jpeg   
4588  images/test/4588.jpeg   
4606  images/test/4606.jpeg   
4608  images/test/4608.jpeg   

                                          product_title    class_label kurta  \
4559  Men Casual Shoe Style Ethnic and Traditional B...  mojaris_women     0   
4564  Stylish Casual Ethnic Nagra/Mojaris/Jutti Form...  mojaris_women     0   
4588           Men's Ethnic Mojari/Juttis/Nagra/Loafers  mojaris_women     0   
4606                 Unique Craft Men's Leather Mojaris  mojaris_women     0   
4608  Wedding Mojaris Shoes for Men Loafer Stylish B...  mojaris_women     0   

     dhoti_pants saree sherwanis palazzos nehru_jackets petticoats blouse  \
4559           0     0         0        0             0          0      0   
4564           0     0         0        0             0          0      0   
4588           0     0         0        0             0          0      0   
4606           0     0         0        0             0          0      0   
4608           0     0         0        0             0          0      0   

     dupattas gowns lehenga women_kurta kurta_men  mojaris_men mojaris_women  \
4559        0     0       0           0         0  mojaris_men             0   
4564        0     0       0           0         0  mojaris_men             0   
4588        0     0       0           0         0  mojaris_men             0   
4606        0     0       0           0         0  mojaris_men             0   
4608        0     0       0           0         0  mojaris_men             0   

     leggings_and_salwars  
4559                    0  
4564                    0  
4588                    0  
4606                    0  
4608                    0

### Many of the images labeled mojaris_women are actually men's

In [352]:
print(f'Product title: {df_mojaris_men.iloc[0,1]}')
print(f'Class label: {df_mojaris_men.iloc[0,2]}')

Product title: Men Casual Shoe Style Ethnic and Traditional Black Leather Jutti Mojari
Class label: mojaris_women


In [353]:
print(f'Product title: {df_mojaris_men.iloc[1,1]}')
print(f'Class label: {df_mojaris_men.iloc[1,2]}')

Product title: Stylish Casual Ethnic Nagra/Mojaris/Jutti Formal Shoes for Men's
Class label: mojaris_women


### Women's Mojaris

In [354]:
df_mojaris_women = df_revised[df_revised.mojaris_women == 'mojaris_women']
df_mojaris_women.shape

(5, 19)

In [355]:
df_mojaris_women.head()

image_path  \
4035  images/test/4035.jpeg   
4112  images/test/4112.jpeg   
4302  images/test/4302.jpeg   
4355  images/test/4355.jpeg   
4464  images/test/4464.jpeg   

                                          product_title  class_label kurta  \
4035  Juttiwala Women's Ethnic Mojari, Juti, Bellies...  mojaris_men     0   
4112                             Women's Fabric Mojaris  mojaris_men     0   
4302              Women Mojari Handicraft Slipper Shoes  mojaris_men     0   
4355  Juttiwala Women's Traditional Slip-On Mojari J...  mojaris_men     0   
4464  Ballet Flats Slip On Mojari Sandal Indian Jutt...  mojaris_men     0   

     dhoti_pants saree sherwanis palazzos nehru_jackets petticoats blouse  \
4035           0     0         0        0             0          0      0   
4112           0     0         0        0             0          0      0   
4302           0     0         0        0             0          0      0   
4355           0     0         0        0             0          0      0   
4464           0     0         0        0             0          0      0   

     dupattas gowns lehenga women_kurta kurta_men mojaris_men  mojaris_women  \
4035        0     0       0           0         0           0  mojaris_women   
4112        0     0       0           0         0           0  mojaris_women   
4302        0     0       0           0         0           0  mojaris_women   
4355        0     0       0           0         0           0  mojaris_women   
4464        0     0       0           0         0           0  mojaris_women   

     leggings_and_salwars  
4035                    0  
4112                    0  
4302                    0  
4355                    0  
4464                    0

### Many of the images labeled mojaris_men are actually women's

In [356]:
print(f'Product title: {df_mojaris_women.iloc[0,1]}')
print(f'Class label: {df_mojaris_women.iloc[0,2]}')

Product title: Juttiwala Women's Ethnic Mojari, Juti, Bellies, Morpankh Design for Woman & Girl Bellies
Class label: mojaris_men


In [357]:
print(f'Product title: {df_mojaris_women.iloc[1,1]}')
print(f'Class label: {df_mojaris_women.iloc[1,2]}')

Product title: Women's Fabric Mojaris
Class label: mojaris_men


### Nehru Jackets

In [358]:
df_nehru_jackets = df_revised[df_revised.nehru_jackets == 'nehru_jackets']
df_nehru_jackets.shape

(17, 19)

In [359]:
df_nehru_jackets.head()

image_path  \
645    images/test/645.jpeg   
2502  images/test/2502.jpeg   
2525  images/test/2525.jpeg   
2534  images/test/2534.jpeg   
2551  images/test/2551.jpeg   

                                          product_title class_label kurta  \
645   KULLU Jacket Nehru Jacket Reversal Both Side U...      blouse     0   
2502  Men Pink &amp; Maroon Solid Kurta with Churida...   kurta_men     0   
2525  Men Teal &amp; Off-White Solid Kurta with Chur...   kurta_men     0   
2534  Men Off-White &amp; Peach-Coloured Solid Kurta...   kurta_men     0   
2551  Printed Full Sleeves Kurta and Churidar Set wi...   kurta_men     0   

     dhoti_pants saree sherwanis palazzos  nehru_jackets petticoats blouse  \
645            0     0         0        0  nehru_jackets          0      0   
2502           0     0         0        0  nehru_jackets          0      0   
2525           0     0         0        0  nehru_jackets          0      0   
2534           0     0         0        0  nehru_jackets          0      0   
2551           0     0         0        0  nehru_jackets          0      0   

     dupattas gowns lehenga women_kurta kurta_men mojaris_men mojaris_women  \
645         0     0       0           0         0           0             0   
2502        0     0       0           0         0           0             0   
2525        0     0       0           0         0           0             0   
2534        0     0       0           0         0           0             0   
2551        0     0       0           0         0           0             0   

     leggings_and_salwars  
645                     0  
2502                    0  
2525                    0  
2534                    0  
2551                    0

### Mislabeled:

In [360]:
print(f'Product title: {df_nehru_jackets.iloc[0,1]}')
print(f'Class label: {df_nehru_jackets.iloc[0,2]}')

Product title: KULLU Jacket Nehru Jacket Reversal Both Side USABLE for Men ONE Side Grey Other Side KINGRY Black
Class label: blouse


### Palazzos

In [361]:
df_palazzos = df_revised[df_revised.palazzos == 'palazzos']
df_palazzos.shape

(182, 19)

In [362]:
df_palazzos.head()

image_path  \
1018  images/test/1018.jpeg   
1034  images/test/1034.jpeg   
1041  images/test/1041.jpeg   
1057  images/test/1057.jpeg   
1091  images/test/1091.jpeg   

                                          product_title  class_label kurta  \
1018  Women's Regular Fit Full Length, Design for Mu...  dhoti_pants     0   
1034  Women Rayon Flared Garara/Sharara Palazzo Pant...  dhoti_pants     0   
1041  Women's Regular Fit Full Length, Design for Mu...  dhoti_pants     0   
1057  R & B Enterprises Stylish Solid Pink Palazzo f...  dhoti_pants     0   
1091               Women's Viscose Palazzo Bottom Pants  dhoti_pants     0   

     dhoti_pants saree sherwanis  palazzos nehru_jackets petticoats blouse  \
1018           0     0         0  palazzos             0          0      0   
1034           0     0         0  palazzos             0          0      0   
1041           0     0         0  palazzos             0          0      0   
1057           0     0         0  palazzos             0          0      0   
1091           0     0         0  palazzos             0          0      0   

     dupattas gowns lehenga women_kurta kurta_men mojaris_men mojaris_women  \
1018        0     0       0           0         0           0             0   
1034        0     0       0           0         0           0             0   
1041        0     0       0           0         0           0             0   
1057        0     0       0           0         0           0             0   
1091        0     0       0           0         0           0             0   

     leggings_and_salwars  
1018                    0  
1034                    0  
1041                    0  
1057                    0  
1091                    0

In [363]:
print(f'Product title: {df_palazzos.iloc[2,1]}')
print(f'Class label: {df_palazzos.iloc[2,2]}')

Product title: Women's Regular Fit Full Length, Design for Multicolor Plain Solid Dhoti,Haram Pant Bottom Wear With Beautiful Flared Sarara/Garara/Palazzo/Free Size Combo (Pack Of 2)
Class label: dhoti_pants


### Petticoats

In [364]:
df_petticoats = df_revised[df_revised.petticoats == 'petticoats']
df_petticoats.shape

(1, 19)

In [365]:
df_petticoats.head()

image_path                                      product_title  \
432  images/test/432.jpeg  Women's Rani Pink Lycra Shapewear Saree Petticoat   

    class_label kurta dhoti_pants saree sherwanis palazzos nehru_jackets  \
432       saree     0           0     0         0        0             0   

     petticoats blouse dupattas gowns lehenga women_kurta kurta_men  \
432  petticoats      0        0     0       0           0         0   

    mojaris_men mojaris_women leggings_and_salwars  
432           0             0                    0

### Mislabeled:

In [366]:
print(f'Product title: {df_petticoats.iloc[0,1]}')
print(f'Class label: {df_petticoats.iloc[0,2]}')

Product title: Women's Rani Pink Lycra Shapewear Saree Petticoat
Class label: saree


### Sherwanis

In [367]:
df_sherwanis = df_revised[df_revised.sherwanis == 'sherwanis']
df_sherwanis.shape

(50, 19)

In [368]:
df_sherwanis.head()

image_path  \
2686  images/test/2686.jpeg   
2726  images/test/2726.jpeg   
4008  images/test/4008.jpeg   
4018  images/test/4018.jpeg   
4048  images/test/4048.jpeg   

                                          product_title  class_label kurta  \
2686                 Men's Silk Sherwani Straight Kurta    kurta_men     0   
2726  Men's Jacquard Silk Floral Print Indowestern S...    kurta_men     0   
4008           Black Velvet Wedding & Sherwani Mojari's  mojaris_men     0   
4018  Jutti for Men's Mens Mojari Indian Shoes Handm...  mojaris_men     0   
4048  Men Royal Luxury & Traditional Designing Handc...  mojaris_men     0   

     dhoti_pants saree  sherwanis palazzos nehru_jackets petticoats blouse  \
2686           0     0  sherwanis        0             0          0      0   
2726           0     0  sherwanis        0             0          0      0   
4008           0     0  sherwanis        0             0          0      0   
4018           0     0  sherwanis        0             0          0      0   
4048           0     0  sherwanis        0             0          0      0   

     dupattas gowns lehenga women_kurta kurta_men mojaris_men mojaris_women  \
2686        0     0       0           0         0           0             0   
2726        0     0       0           0         0           0             0   
4008        0     0       0           0         0           0             0   
4018        0     0       0           0         0           0             0   
4048        0     0       0           0         0           0             0   

     leggings_and_salwars  
2686                    0  
2726                    0  
4008                    0  
4018                    0  
4048                    0

### Mislabeled: It is a sherwani-style kurta, not a sherwani

In [369]:
print(f'Product title: {df_sherwanis.iloc[0,1]}')
print(f'Class label: {df_sherwanis.iloc[0,2]}')

Product title: Men's Silk Sherwani Straight Kurta
Class label: kurta_men


### Women's Kurtas

In [370]:
df_women_kurta = df_revised[df_revised.women_kurta == 'women_kurta']
df_women_kurta.shape

(169, 19)

In [371]:
df_women_kurta.head()

image_path  \
111    images/test/111.jpeg   
1002  images/test/1002.jpeg   
1016  images/test/1016.jpeg   
1027  images/test/1027.jpeg   
1028  images/test/1028.jpeg   

                                          product_title  class_label kurta  \
111   Women's georgette with blouse piece Kurta (Kur...        saree     0   
1002  Women Cotton Floral and Embroidered Pattern St...  dhoti_pants     0   
1016  Women's Black Pure Cotton Straight Kurta With ...  dhoti_pants     0   
1027  Rayon Kurta and Multicolor Dhoti Pant Set For ...  dhoti_pants     0   
1028  Women's Sage Green Umbrella Print Rayon Kurta ...  dhoti_pants     0   

     dhoti_pants saree sherwanis palazzos nehru_jackets petticoats blouse  \
111            0     0         0        0             0          0      0   
1002           0     0         0        0             0          0      0   
1016           0     0         0        0             0          0      0   
1027           0     0         0        0             0          0      0   
1028           0     0         0        0             0          0      0   

      dupattas gowns lehenga  women_kurta kurta_men mojaris_men mojaris_women  \
111          0     0       0  women_kurta         0           0             0   
1002  dupattas     0       0  women_kurta         0           0             0   
1016         0     0       0  women_kurta         0           0             0   
1027         0     0       0  women_kurta         0           0             0   
1028         0     0       0  women_kurta         0           0             0   

     leggings_and_salwars  
111                     0  
1002                    0  
1016                    0  
1027                    0  
1028                    0

In [372]:
print(f'Product title: {df_women_kurta.iloc[0,1]}')
print(f'Class label: {df_women_kurta.iloc[0,2]}')

Product title: Women's georgette with blouse piece Kurta (Kurtas321_ Rani_ 6 Meter)
Class label: saree


In [373]:
print(f'Product title: {df_women_kurta.iloc[1,1]}')
print(f'Class label: {df_women_kurta.iloc[1,2]}')

Product title: Women Cotton Floral and Embroidered Pattern Straight Kurta and Pant with Dupatta Set (STYLEP)
Class label: dhoti_pants


## While not all of the items in each dataframe are mislabeled, many are. 
## There are 191 rows in the saree dataframe;
## 201 rows in the blouse dataframe;
## 45 rows in the dhoti_pants dataframe;
## 278 rows in the dupatta dataframe;
## 174 rows in the gown dataframe;
## 16 rows in the kurta dataframe;
## 61 rows in the kurta_men dataframe;
## 214 rows in the leggings_and_salwars dataframe;
## 29 rows in the lehenga dataframe;
## 16 rows in the mojaris_men dataframe;
## 5 rows in the mojaris_women dataframe;
## 17 rows in the Nehru_jackets dataframe;
## 182 rows in the palazzos dataframe;
## 1 row in the petticoats dataframe;
## 50 rows in the sherwani dataframe;
## and 169 rows in the women_kurta dataframe.